### ISSUE TITLE
Bug: cache is not working when running on GitHub actions

### ISSUE THREAD

**Issue by AntonioVentilii** [eslint/eslint#20101]
### Environment

Node version: 22.14.0
npm version: 10.9.2
Local ESLint version: 9.32.0
Global ESLint version: 9.32.0
Operating System: ubuntu-24.04 (Github Actions Runner)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

Any configuration, minimal.

```yaml
- name: ESLint cache
  uses: actions/cache@v3
  with:
    path: .eslintcache
    key: |
      ${{ runner.os }}-eslint-${{ github.event.pull_request.number }}
    restore-keys: |
      ${{ runner.os }}-eslint-

- name: Run ESLint
  run: eslint --cache --debug .
```

### What did you expect to happen?

I would expect the Github action to run the second time faster, using the cache.

### What actually happened?

ESLint says it has picked up the configuration:

```
eslint:lint-result-cache Caching results to /home/runner/work/[...].eslintcache
```

but for every file, the log will show (even without changes in files):

```
eslintrc:cascading-config-array-factory Cache hit: /home/runner/work/[...]/dir.
eslint:lint-result-cache Cache entry not found or no longer valid: /home/runner/work/[...]/dir/file.ext
```

### Link to Minimal Reproducible Example

You can test it in any repo you want, since it is Github Actions.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

There was [a question raised already in Discussions](https://github.com/eslint/eslint/discussions/16997), but it received no satisfactory answer (or at least the answer is not valid anymore).

**Comment by AntonioVentilii** [comment]
So, using `--cache-strategy content` just for the CIs solves it:

```
eslint --cache ${CI:+--cache-strategy content} .
```

ESLint is invalidating every entry because file `mtimes` change on each CI checkout. By default ESLint’s cache uses the `metadata` strategy (`mtime` + `size`), so a fresh checkout makes it look like “everything changed”.

# `metadata` (the default)

ESLint caches based on file `size` + `mtime`.

Works well on a dev machine, because `mtime` stays consistent.

On CI, every fresh `git checkout` resets `mtimes` → cache invalidated → almost useless.

# `content`

ESLint hashes the file contents instead of relying on mtime.

Identical files across runs keep their cache entries.

Much more reliable in ephemeral environments (CI/CD, Docker builds, etc.).
